modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/CheckboxTag.java

```diff
@@ -0,0 +1,192 @@
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Kresimir Coko
+ */
+public class CheckboxTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		return super.doStartTag();
+	}
+
+	public boolean getChecked() {
+		return _checked;
+	}
+
+	public boolean getDisabled() {
+		return _disabled;
+	}
+
+	public boolean getIndeterminate() {
+		return _indeterminate;
+	}
+
+	public boolean getInline() {
+		return _inline;
+	}
+
+	public String getLabel() {
+		return _label;
+	}
+
+	public String getName() {
+		return _name;
+	}
+
+	public void setChecked(boolean checked) {
+		_checked = checked;
+	}
+
+	public void setDisabled(boolean disabled) {
+		_disabled = disabled;
+	}
+
+	public void setIndeterminate(boolean indeterminate) {
+		_indeterminate = indeterminate;
+	}
+
+	public void setInline(boolean inline) {
+		_inline = inline;
+	}
+
+	public void setLabel(String label) {
+		_label = label;
+	}
+
+	public void setName(String name) {
+		_name = name;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_checked = false;
+		_disabled = false;
+		_indeterminate = false;
+		_inline = false;
+		_label = null;
+		_name = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/Checkbox";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("checked", _checked);
+		props.put("disabled", _disabled);
+		props.put("indeterminate", _indeterminate);
+		props.put("inline", _inline);
+
+		if (Validator.isNotNull(_label)) {
+			props.put(
+				"label",
+				LanguageUtil.get(
+					TagResourceBundleUtil.getResourceBundle(pageContext),
+					_label));
+		}
+
+		props.put("name", _name);
+
+		return super.prepareProps(props);
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		cssClasses.add("custom-control");
+		cssClasses.add("custom-checkbox");
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		jspWriter.write("<label><input");
+
+		if (_checked) {
+			jspWriter.write(" checked");
+		}
+
+		jspWriter.write(" class=\"custom-control-input ");
+
+		if (Validator.isNotNull(getCssClass())) {
+			jspWriter.write(getCssClass());
+		}
+
+		jspWriter.write("\"");
+
+		if (_disabled) {
+			jspWriter.write(" disabled");
+		}
+
+		if (Validator.isNotNull(getId())) {
+			jspWriter.write(" id=\"");
+			jspWriter.write(getId());
+			jspWriter.write("\"");
+		}
+
+		if (Validator.isNotNull(_name)) {
+			jspWriter.write(" name=\"");
+			jspWriter.write(_name);
+			jspWriter.write("\"");
+		}
+
+		jspWriter.write(" type=\"checkbox\" />");
+
+		if (Validator.isNotNull(_label)) {
```

Comment:
It's a bit weird that we'd unconditionally add the `<label>` tag even when we don't have a `_label`... is that really necessary for the markup to behave as expected? It may well be, but want to double check.

Reply:
That's a good point, it's probably related to the above comment too https://github.com/liferay-frontend/liferay-portal/pull/427#discussion_r511897532

Reply:
This is replicating exactly what [Clay Checkbox](https://github.com/liferay/clay/blob/aefb430bd83de2056122ae2af3fb06cae0b2f6ae/packages/clay-form/src/Checkbox.tsx#L64) renders, and IMHO that's what we should do, even if "we don't need this if we don't have a label". I think that would be something to fix directly in Clay (and then update it here), but we need to keep both clay and tag in sync. It's difficult enough the fact that we need to duplicate the markup to not duplicate it but re-write it.

Reply:
Yeah, that's okay, just making sure 👍

Reply:
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia1.giphy.com%2Fmedia%2Fl0HUqsz2jdQYElRm0%2Fgiphy.gif&token=dance" data-gitmeme-token="dance"><img src="https://media1.giphy.com/media/l0HUqsz2jdQYElRm0/giphy.gif" title="Created by gitme.me with /dance"/></a>